rdmd_test.d

```diff
@@ -355,24 +357,107 @@ void runTests()
     assert(res.status == 0, res.output ~ "\nCan't run with --compiler=" ~ fullCompilerPath);
     assert(res.output.canFind("compile_force_src"));
 
-    /* tmpdir */
+    // Create an empty temporary directory and clean it up when exiting scope
+    static struct TmpDir
+    {
+        string name;
+        this(string name)
+        {
+            this.name = name;
+            if (exists(name)) rmdirRecurse(name);
+            mkdir(name);
+        }
+        @disable this(this);
+        ~this()
+        {
+            import core.thread;
+            Thread.sleep(100.msecs); // Hack around Windows locking the directory
```

Comment:
Put a version around it? Are there better workarounds? We shouldn't have people wait 100 extra milliseconds on all platforms. Note that currently "rdmd hello.d" takes 200ms on a typical Posix system to build.

I'll ask for changes because of this, rest are optional nits.